{"abstract":[{"text":"The method to call when a critical error occurs.","type":"text"}],"primaryContentSections":[{"declarations":[{"platforms":["iOS"],"tokens":[{"text":"var","kind":"keyword"},{"text":" ","kind":"text"},{"text":"apptentiveAssertionHandler","kind":"identifier"},{"text":": (","kind":"text"},{"text":"@autoclosure ","kind":"attribute"},{"text":"() -> ","kind":"text"},{"preciseIdentifier":"s:SS","text":"String","kind":"typeIdentifier"},{"text":", ","kind":"text"},{"preciseIdentifier":"s:s12StaticStringV","kind":"typeIdentifier","text":"StaticString"},{"kind":"text","text":", "},{"kind":"typeIdentifier","preciseIdentifier":"s:Su","text":"UInt"},{"kind":"text","text":") -> ()"}],"languages":["swift"]}],"kind":"declarations"},{"content":[{"anchor":"discussion","text":"Discussion","type":"heading","level":2},{"inlineContent":[{"type":"text","text":"This can be overriden, for example:"}],"type":"paragraph"},{"syntax":null,"type":"codeListing","code":["apptentiveAssertionHandler = { message, file, line in","    print(\"\\(file):\\(line): Apptentive critical error: \\(message())\")","}"]}],"kind":"content"}],"identifier":{"url":"doc:\/\/com.apptentive.Apptentive\/documentation\/ApptentiveKit\/apptentiveAssertionHandler","interfaceLanguage":"swift"},"variants":[{"paths":["\/documentation\/apptentivekit\/apptentiveassertionhandler"],"traits":[{"interfaceLanguage":"swift"}]}],"metadata":{"roleHeading":"Global Variable","title":"apptentiveAssertionHandler","symbolKind":"var","role":"symbol","modules":[{"name":"ApptentiveKit"}],"fragments":[{"kind":"keyword","text":"var"},{"text":" ","kind":"text"},{"text":"apptentiveAssertionHandler","kind":"identifier"},{"kind":"text","text":": (() -> "},{"preciseIdentifier":"s:SS","text":"String","kind":"typeIdentifier"},{"text":", ","kind":"text"},{"text":"StaticString","preciseIdentifier":"s:s12StaticStringV","kind":"typeIdentifier"},{"kind":"text","text":", "},{"preciseIdentifier":"s:Su","text":"UInt","kind":"typeIdentifier"},{"kind":"text","text":") -> ()"}],"externalID":"s:13ApptentiveKit26apptentiveAssertionHandleryySSyXK_s12StaticStringVSutcvp"},"schemaVersion":{"minor":3,"major":0,"patch":0},"hierarchy":{"paths":[["doc:\/\/com.apptentive.Apptentive\/documentation\/ApptentiveKit"]]},"sections":[],"kind":"symbol","references":{"doc://com.apptentive.Apptentive/documentation/ApptentiveKit/apptentiveAssertionHandler":{"type":"topic","identifier":"doc:\/\/com.apptentive.Apptentive\/documentation\/ApptentiveKit\/apptentiveAssertionHandler","url":"\/documentation\/apptentivekit\/apptentiveassertionhandler","title":"apptentiveAssertionHandler","kind":"symbol","role":"symbol","fragments":[{"kind":"keyword","text":"var"},{"kind":"text","text":" "},{"text":"apptentiveAssertionHandler","kind":"identifier"},{"kind":"text","text":": (() -> "},{"kind":"typeIdentifier","preciseIdentifier":"s:SS","text":"String"},{"text":", ","kind":"text"},{"preciseIdentifier":"s:s12StaticStringV","text":"StaticString","kind":"typeIdentifier"},{"text":", ","kind":"text"},{"preciseIdentifier":"s:Su","kind":"typeIdentifier","text":"UInt"},{"text":") -> ()","kind":"text"}],"abstract":[{"type":"text","text":"The method to call when a critical error occurs."}]},"doc://com.apptentive.Apptentive/documentation/ApptentiveKit":{"identifier":"doc:\/\/com.apptentive.Apptentive\/documentation\/ApptentiveKit","type":"topic","title":"ApptentiveKit","kind":"symbol","url":"\/documentation\/apptentivekit","role":"collection","abstract":[]}}}